shuup/core/models/_shipments.py

```diff
@@ -156,21 +155,26 @@ def set_received(self, purchase_prices=None, created_by=None):
                     created_by=created_by)
 
 
+# TODO (2.0): Rename this to OrderShipment
+class Shipment(AbstractShipment):
+    order = models.ForeignKey(
+        "Order", blank=True, null=True, related_name='shipments', on_delete=models.PROTECT,
+        verbose_name=_("order"))
+    supplier = models.ForeignKey(
+        "Supplier", related_name='shipments', on_delete=models.PROTECT, verbose_name=_("supplier"))
+
+
+# TODO (2.0): Rename this to ShipmentProduct
 @python_2_unicode_compatible
-class ShipmentProduct(models.Model):
-    shipment = models.ForeignKey(
-        Shipment, related_name='products', on_delete=models.PROTECT, verbose_name=_("shipment")
-    )
-    product = models.ForeignKey(
-        "Product", related_name='shipments', on_delete=models.CASCADE, verbose_name=_("product")
-    )
```

Comment:
I don't understand why product needs to be taken away from the abstract base class. (Shipment is separated clearly, because if you subclass AbstractShipment, then you probably want to have subclass of AbstractShipmentProduct for its products, but it would still have products. Right?)